pyqtorch/utils.py

```diff
@@ -125,6 +125,10 @@ def param_dict(keys: Sequence[str], values: Sequence[Tensor]) -> dict[str, Tenso
     return {key: val for key, val in zip(keys, values)}
 
 
+class DensityMatrix(Tensor):
```

Comment:
Is it necessary to define a custom type if it does pretty much nothing ?

Reply:
The `DensityMatrix` class is used as a "flag" to ensure the condition in the different `forward` functions. :
```python3
if not isinstance(state, DensityMatrix):
```
It allows to check if the input state is a vector state or density matrix without any condition on the `size`, `n_qubits`... So `if not isinstance(state, DensityMatrix):` just check a `bool` right?

Reply:
Ah right. OK I see.